packages/codec/lib/compilations/utils.ts

```diff
@@ -398,51 +398,52 @@ function inferLanguage(
 
 function getIndexToAddAt(
   sourceObject: Source,
-  index: number,
+  index: number | undefined,
   sources: Source[],
   unreliableSourceOrder: boolean
 ): { index: number; needsAdding: boolean; unreliableSourceOrder: boolean } {
-  //first: is this already there? only add it if it's not.
-  //(we determine this by sourcePath if present, and the actual source
-  //contents if not)
   debug("sourcePath: %s", sourceObject.sourcePath);
   debug("given index: %d", index);
   debug(
     "sources: %o",
     sources.map(source => source.sourcePath)
   );
-  if (
-    sources.every(
-      existingSource =>
-        existingSource.sourcePath !== sourceObject.sourcePath ||
+  //first: is this already there? only add it if it's not.
+  //(we determine this by sourcePath if present, and the actual source
+  //contents if not)
+  const existingIndex = sources.findIndex(
+    existingSource =>
+      existingSource &&
```

Comment:
Is it possible for an element of  `sources` to be false-y?

Reply:
Sort of.  No actual element will be falsy, but the array may be sparse.  Different array functions differ in whether they just skip non-existent elements, or treat them as `undefined`; `findIndex`, it seems, does the latter, which is why I had to add the guard.

Reply:
That's good to know! It's worth leaving a comment to explain the guard.

Reply:
OK, I can add that.

Reply:
Added that!